include/BmsDataTypes.hpp

```diff
@@ -0,0 +1,117 @@
+constexpr uint16_t BMS_ERR_STATUS_DSG_OCP       {utils::bitIdxToValue<uint16_t>(BMS_ERR_STATUS_DSG_OCP_BIT_IDX)};       //    512 - bit9  Discharge overcurrent protection
+constexpr uint16_t BMS_ERR_STATUS_SHORT_CIRCUIT {utils::bitIdxToValue<uint16_t>(BMS_ERR_STATUS_SHORT_CIRCUIT_BIT_IDX)}; //   1024 - bit10 short circuit protection
+constexpr uint16_t BMS_ERR_STATUS_AFE_ERROR     {utils::bitIdxToValue<uint16_t>(BMS_ERR_STATUS_AFE_ERROR_BIT_IDX)};     //   2048 - bit11 Front-end detection IC error
+constexpr uint16_t BMS_ERR_STATUS_SOFT_LOCK     {utils::bitIdxToValue<uint16_t>(BMS_ERR_STATUS_SOFT_LOCK_BIT_IDX)};     //   4096 - bit12 software lock MOS
+constexpr uint16_t BMS_ERR_STATUS_RESERVED1     {utils::bitIdxToValue<uint16_t>(BMS_ERR_STATUS_RESERVED1_BIT_IDX)};     //   8192 - bit13 Reserved
+constexpr uint16_t BMS_ERR_STATUS_RESERVED2     {utils::bitIdxToValue<uint16_t>(BMS_ERR_STATUS_RESERVED2_BIT_IDX)};     //  16384 - bit14 Reserved
+constexpr uint16_t BMS_ERR_STATUS_RESERVED3     {utils::bitIdxToValue<uint16_t>(BMS_ERR_STATUS_RESERVED3_BIT_IDX)};     //  32768 - bit15 Reserved
+
+/**
+ * @brief This method convert the JkBmsWarnMsg into the BmsErrorStatus.
+ * */
+inline BmsErrorStatus bmsErrorFromMessage(const jkbms::JkBmsWarnMsg& bmsMsg)
+{
+  using namespace jkbms;
+  BmsErrorStatus bmsStat = BmsErrorStatus::from_int(BMS_ERR_STATUS_OK);
+                                                                                                          // Bit  0: Low capacity alarm
+  if(bmsMsg.is_set(BatWarnMsgBits::MOS_TUBE_OVERTEMP_ALARM))    { bmsStat |= BmsErrorBits::CHG_OTP; }     // Bit  1: MOS tube over temperature alarm            -> ?
+  if(bmsMsg.is_set(BatWarnMsgBits::CHG_OVERVOLTAGE_ALARM))      { bmsStat |= BmsErrorBits::BATTERY_OVP; } // Bit  2: Charge over voltage alarm                  -> ?
+  if(bmsMsg.is_set(BatWarnMsgBits::CELL_OVERVOLTAGE))           { bmsStat |= BmsErrorBits::CELL_OVP; }    // Bit  3: cell over voltage                          -> x
+  if(bmsMsg.is_set(BatWarnMsgBits::CELL_UNDERVOLTAGE))          { bmsStat |= BmsErrorBits::CELL_UVP; }    // Bit  4: cell under voltage                         -> x
+  if(bmsMsg.is_set(BatWarnMsgBits::CHG_OVERTEMP))               { bmsStat |= BmsErrorBits::CHG_OTP; }     // Bit  5: Charge over temperature                    -> x
+  if(bmsMsg.is_set(BatWarnMsgBits::DCHG_OVERCURRENT_ALARM))     { bmsStat |= BmsErrorBits::DCHG_OCP; }    // Bit  6: discharge over current alarm               -> ?
+  if(bmsMsg.is_set(BatWarnMsgBits::DCHG_OVERCURRENT))           { bmsStat |= BmsErrorBits::DCHG_OCP; }    // Bit  7: discharge overcurent                       -> x
+  if(bmsMsg.is_set(BatWarnMsgBits::BATTERY_BOX_OVERTEMP_ALARM)) { bmsStat |= BmsErrorBits::CHG_OTP; }     // Bit  8: over temperature alarm in the battery box  -> ?
+  // TODO : Is it correct, that BATTERY_LOW_TEMPERATURE is mapped to CHG_OCP (was Bit 9:  Battery low temperature to BMS_ERR_STATUS_CHG_OCP)
```

Comment:
@shining-man Can you please check, if this is correct

Reply:
Is ok